### PR TITLE
feat: add check to open overlay by clicking on icon

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
Add query parameter to show overlay or redirection 

Fixes https://github.com/devtron-labs/sprint-tasks/issues/1408

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/externalLinks/ExternalLinks.component.tsx [hunk 1]
```diff
@@ -201,6 +202,8 @@ const ExternalLinkChip = ({ linkOption, idx, handleOpenModal, details, isOvervie
     const externalLinkURL = getParsedURL(true, linkOption.value.toString(), details)
     const handleTextClick = () => handleOpenModal(linkOption, externalLinkURL)
 
+    const openOverlayOnIcon: boolean = `${linkOption.value}`.includes('devtronIframePrimary=false')
```
Comment (AbhishekA1509) [inline src/components/externalLinks/ExternalLinks.component.tsx:205]: Please parse params and check if exists and move this to constant

File: src/components/externalLinks/ExternalLinks.component.tsx [hunk 2]
```diff
@@ -224,7 +227,11 @@ const ExternalLinkChip = ({ linkOption, idx, handleOpenModal, details, isOvervie
                 <button
                     className="flexbox dc__gap-4 px-6 py-2 dc__align-items-center dc__unset-button-styles dc__hover-n50 dc__left-radius-3"
                     type="button"
-                    onClick={handleTextClick}
+                    onClick={
+                        openOverlayOnIcon
+                            ? getHandleOpenURL(externalLinkURL)
```
Comment (AbhishekA1509) [inline src/components/externalLinks/ExternalLinks.component.tsx:232]: Please santize externalLinkURL by removing devtronIframePrimary

File: src/components/externalLinks/ExternalLinks.component.tsx [hunk 3]
```diff
@@ -236,12 +243,16 @@ const ExternalLinkChip = ({ linkOption, idx, handleOpenModal, details, isOvervie
                 </button>
                 <a
                     key={linkOption.label}
-                    href={getParsedURL(true, linkOption.value.toString(), details)}
-                    target="_blank"
-                    className="flex p-4 open-link-button dc__hover-n50 dc__right-radius-3"
-                    rel="noreferrer"
+                    {...(openOverlayOnIcon
+                        ? { onClick: handleTextClick }
```
Comment (AbhishekA1509) [inline src/components/externalLinks/ExternalLinks.component.tsx:247]: Ideally a tag should contain href

File: src/components/externalLinks/ExternalLinksCRUD/AddExternalLink.tsx [hunk 4]
```diff
@@ -320,12 +349,30 @@ export default function AddExternalLink({
...
-                    </ul>
-                    <ExternalLinksLearnMore />
+                    </div>
+                    <li className="pt-16">Open link in new tab or overlay</li>
+                    <div className="fs-13 lh-20 fw-4 flexbox-col dc__gap-4">
+                        <span>
+                            <span>
+                                Links open in an overlay by default. To open the link in a new tab by default, add
+                            </span>
+                            &nbsp;<span className="bcn-1 px-4 br-6">devtronIframePrimary=false</span>&nbsp;
```
Comment (AbhishekA1509) [inline src/components/externalLinks/ExternalLinksCRUD/AddExternalLink.tsx:368]: Please re-use the constant that we will define now

File: src/components/externalLinks/ExternalLinksCRUD/AddExternalLink.tsx [hunk 5]
```diff
@@ -320,12 +349,30 @@ export default function AddExternalLink({
                     <p className="mb-12 fw-4 dc__italic-font-style">
                         {`http://www.domain.com/{namespace}/{appName}/details/{appId}/env/{envId}/details/{podName}`}
                     </p>
-                    <ul className="available-link-variables fs-12 fw-4">
-                        {availableVariables.map((_var) => (
-                            <li key={_var}>{_var}</li>
+                    <div className="flexbox-col dc__gap-6">
+                        {availableVariables.map((variable) => (
+                            <div key={variable} className="flexbox dc__gap-4 dc__align-items-center dc__visible-hover dc__visible-hover--parent">
```
Comment (AbhishekA1509) [inline src/components/externalLinks/ExternalLinksCRUD/AddExternalLink.tsx:354]: Check if truncate or word-break is required

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/components/externalLinks/ExternalLinks.component.tsx [hunk 6]
```diff
@@ -198,9 +200,17 @@ const ExternalLinkIframeModal = ({ selectedExternalLink, handleCloseModal }) =>
 )
 
 const ExternalLinkChip = ({ linkOption, idx, handleOpenModal, details, isOverviewPage }: ExternalLinkChipProps) => {
-    const externalLinkURL = getParsedURL(true, linkOption.value.toString(), details)
+    let externalLinkURL = getParsedURL(true, linkOption.value.toString(), details)
     const handleTextClick = () => handleOpenModal(linkOption, externalLinkURL)
 
+    const [baseUrl, searchString] = externalLinkURL.split('?')
```
Comment (AbhishekA1509) [inline src/components/externalLinks/ExternalLinks.component.tsx:206]: Ideally should be done in service itself

**Review by AbhishekA1509 — DISMISSED** [review]

**Review by AbhishekA1509 — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**